web/src/peer_data.ts

```diff
@@ -251,7 +252,10 @@ export function add_subscriber(stream_id: number, user_id: number): void {
     if (person === undefined) {
         blueslip.warn(`We tried to add invalid subscriber: ${user_id}`);
     }
-    subscribers.add(user_id);
+    if (!subscribers.has(user_id)) {
+        subscribers.add(user_id);
+        sub_store.get(stream_id)!.subscriber_count += 1;
+    }
```

Comment:
I should add tests for this logic, haven't gotten to that yet

Reply:
Oh if I update `get_subscriber_count` to use `subscriber_count` then I think other tests will already cover this, sweet.